### PR TITLE
add classes Machine, Truck, Bulldozer, Excavato and their implementation

### PR DESCRIPTION
**Pull request by mkazarian-1**

### CODE REVIEW

File: src/main/java/core/basesyntax/Machine.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public abstract class Machine {
+
+    public abstract void doWork();
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Machine.java:5]:
```suggestion
public abstract class Machine {
    public abstract void doWork();
```

File: src/main/java/core/basesyntax/MainApp.java [hunk 2]
```diff
@@ -1,5 +1,16 @@
 package core.basesyntax;
 
 public class MainApp {
+    public static void main(String[] args) {
 
+        Machine truck = new Truck();
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/MainApp.java:6]: image

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/MainApp.java (hunk 2 above)
Comment (Rommelua) [reply src/main/java/core/basesyntax/MainApp.java:6]: Line No 5 is still empty. Do not begin methods with empty line.

File: src/main/java/core/basesyntax/MainApp.java [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+        Machine excavator = new Excavator();
+        Machine buildozer = new Bulldozer();
+        Machine[] array = new Machine[]{truck, excavator, buildozer};
+
+        for (Machine x : array) {
+            x.doWork();
+            x.stopWork();
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/MainApp.java:13]:
```suggestion
        for (Machine machine : machines) {
            machine .doWork();
            machine .stopWork();
```

File: src/main/java/core/basesyntax/MainApp.java [hunk 4]
```diff
@@ -1,5 +1,16 @@
 package core.basesyntax;
 
 public class MainApp {
+    public static void main(String[] args) {
 
+        Machine truck = new Truck();
+        Machine excavator = new Excavator();
+        Machine buildozer = new Bulldozer();
+        Machine[] array = new Machine[]{truck, excavator, buildozer};
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/MainApp.java:9]:
```suggestion
        Machine[] machines = new Machine[]{truck, excavator, buildozer};
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Review by sokimaaa — APPROVED** [review]

**Pull request closed without merging**